test/formats/test_asf.py

```diff
@@ -0,0 +1,57 @@
+from test.picardtestcase import (
+    PicardTestCase,
+    create_fake_png,
+)
+
+from picard.formats import asf
+from .common import CommonTests
+from .coverart import CommonCoverArtTests
+
+
+class ASFTest(CommonTests.TagFormatsTestCase):
+    testfile = 'test.asf'
+    supports_ratings = True
+    expected_info = {
+        'length': 92,
+        '~channels': '2',
+        '~sample_rate': '44100',
+        '~bitrate': '128.0',
+    }
+
+
+class WMATest(CommonTests.TagFormatsTestCase):
+    testfile = 'test.wma'
+    supports_ratings = True
+    expected_info = {
+        'length': 139,
+        '~channels': '2',
+        '~sample_rate': '44100',
+        '~bitrate': '64.0',
+    }
+
+
+class AsfUtilTest(PicardTestCase):
+    def test_pack_and_unpack_image(self):
+        mime = 'image/png'
+        image_data = create_fake_png(b'x')
+        type = 4
```

Comment:
I'd rather use another name, as `type` is a [builtin](https://docs.python.org/3/library/functions.html#type). Not critical though.